### PR TITLE
doc: Add initial readme

### PR DESCRIPTION
**Pull request by trygve-lie**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,2 +1,72 @@
-# renovate-presets
-Renovate bot presets
+# Renovate bot presets
+
+This repo contain a set of [Renovate Bot](https://www.whitesourcesoftware.com/free-developer-tools/renovate/) presets modules in this organization can depend on.
```
Comment (digitalsadhu) [inline README.md]:
```suggestion
This repo contain a set of [Renovate Bot](https://www.whitesourcesoftware.com/free-developer-tools/renovate/) presets that modules in this organization can depend on.
```

File: README.md [hunk 2]
```diff
@@ -1,2 +1,72 @@
...
+
+The purpose of these presets is to be able to optimize the process of keeping dependencies up to date accross all modules in this organization while minimizing the amount of releases of each module in this organization caused by automatically bumping dependencies.
+
+As an example lets say we have a set of modules all under Renovate Bot and Semantic Release structured like so:
```
Comment (digitalsadhu) [inline README.md]:
```suggestion
As an example lets say we have a set of modules using Renovate Bot and Semantic Release structured like so:
```

File: README.md [hunk 3]
```diff
@@ -1,2 +1,72 @@
...
+
+```sh
+|--/@eik/module-a
+   |--/3rd-party-dependency
+   |--/@eik/module-b
+      |--/3rd-party-dependency
+```
+
+There is a `@eik/module-a` which depend on `3rd-party-dependency` and a `@eik/module-b`. `@eik/module-b` do also depend on the same `3rd-party-dependency`.
```
Comment (digitalsadhu) [inline README.md]:
```suggestion
`@eik/module-a` and `@eik/module-b` depend on the same `3rd-party-dependency`.
```

File: README.md [hunk 4]
```diff
@@ -1,2 +1,72 @@
...
+
+The `@eik/module-a` is a publicly facing module. Its a module which others will use as a dependency. `@eik/module-b` is __not__ a publicly facing module. Its an internal module which contain code only of interest of other modules in our organization and it should never be used by externals as a dependency.
```
Comment (digitalsadhu) [inline README.md]:
```suggestion
`@eik/module-a` is a public facing module that other modules can use as a dependency. `@eik/module-b` is __not__ a public facing module. Rather, it's an internal module which contains code that is only of interest to other modules within our organisation and it should never be used as a dependency by external modules.
```

File: README.md [hunk 5]
```diff
@@ -1,2 +1,72 @@
...
+
+If we where to use the default configuration from Renovate Bot the following would happen when we have Semantic Release in our setup and there is a new version of the `3rd-party-dependency`: `@eik/module-a` would get a PR with an update of the `3rd-party-dependency` and when merged a new release of `@eik/module-a` would be cut. Meanwhile `@eik/module-b` would also get a PR with an update of the `3rd-party-dependency` and when merged a new release of `@eik/module-b` would also be cut.
```
Comment (digitalsadhu) [inline README.md]:
```suggestion
If we were to use the default configuration from Renovate Bot the following would happen when we have Semantic Release in our setup and there is a new version of the `3rd-party-dependency`: `@eik/module-a` would get a PR with an update of the `3rd-party-dependency` and when merged a new release of `@eik/module-a` would be cut. Meanwhile `@eik/module-b` would also get a PR with an update of the `3rd-party-dependency` and when merged a new release of `@eik/module-b` would also be cut. 
```

File: README.md [hunk 6]
```diff
@@ -1,2 +1,72 @@
...
+
+The `@eik/module-a` is a publicly facing module. Its a module which others will use as a dependency. `@eik/module-b` is __not__ a publicly facing module. Its an internal module which contain code only of interest of other modules in our organization and it should never be used by externals as a dependency.
+
+If we where to use the default configuration from Renovate Bot the following would happen when we have Semantic Release in our setup and there is a new version of the `3rd-party-dependency`: `@eik/module-a` would get a PR with an update of the `3rd-party-dependency` and when merged a new release of `@eik/module-a` would be cut. Meanwhile `@eik/module-b` would also get a PR with an update of the `3rd-party-dependency` and when merged a new release of `@eik/module-b` would also be cut. 
+
+This release of `@eik/module-b` would again create a new PR in `@eik/module-a` and cause a new release of `@eik/module-a` when merged. 
+
+In other words; Due to both `@eik/module-a` and `@eik/module-b` depending on `3rd-party-dependency` an update of `3rd-party-dependency` will cause two releases of `@eik/module-a`. Which is not optimal.
```
Comment (digitalsadhu) [inline README.md]:
```suggestion
In other words; Due to both `@eik/module-a` and `@eik/module-b` depending on `3rd-party-dependency` an update of `3rd-party-dependency` will cause two releases of `@eik/module-a` which is not optimal.
```

File: README.md [hunk 7]
```diff
@@ -1,2 +1,72 @@
...
+
+## Module categorisation
+
+To deal with problems such as described in the introduction above we separate modules into two categories, top modules and sub modules, where each category have slightly different grouping and dependency strategies.
```
Comment (digitalsadhu) [inline README.md]:
```suggestion
To deal with problems such as that described in the introduction above we separate modules into two categories, top level modules and sub modules, where each category has slightly different groupings and dependency strategies.
```

File: README.md [hunk 8]
```diff
@@ -1,2 +1,72 @@
...
+
+In short; sub modules recieve and merge dependencies as they arrive. Top modules recieve and merge dependencies on a less frequent schedule. This more or less removes the above outlined problem.
```
Comment (digitalsadhu) [inline README.md]:
```suggestion
In short; sub modules receive and merge dependencies as they arrive. Top level modules receive and merge dependencies on a less frequent schedule. This more or less removes the problem outlined above.
```

File: README.md [hunk 9]
```diff
@@ -1,2 +1,72 @@
...
+
+### Top modules
```
Comment (digitalsadhu) [inline README.md]:
```suggestion
### Top level modules
```

File: README.md [hunk 10]
```diff
@@ -1,2 +1,72 @@
...
+
+Top modules is publicly facing modules. This is modules which others will use as a dependency.
```
Comment (digitalsadhu) [inline README.md]:
```suggestion
Top level modules are publicly facing modules. These are modules that other external modules and projects will use as a dependency. 
```

File: README.md [hunk 11]
```diff
@@ -1,2 +1,72 @@
...
+
+To deal with problems such as described in the introduction above we separate modules into two categories, top modules and sub modules, where each category have slightly different grouping and dependency strategies.
+
+In short; sub modules recieve and merge dependencies as they arrive. Top modules recieve and merge dependencies on a less frequent schedule. This more or less removes the above outlined problem.
+
+### Top modules
+
+Top modules is publicly facing modules. This is modules which others will use as a dependency. 
+
+Top modules has its dependencies updated once a week.
```
Comment (digitalsadhu) [inline README.md]:
```suggestion
Top level modules have their dependencies updated once a week.
```

File: README.md [hunk 12]
```diff
@@ -1,2 +1,72 @@
...
+
+To use the preset for top modules in a module add the following config to `renovate.json`:
```
Comment (digitalsadhu) [inline README.md]:
```suggestion
To use the preset for top level modules within another module, add the following config to `renovate.json`:
```

File: README.md [hunk 13]
```diff
@@ -1,2 +1,72 @@
...
+  "$schema": "https://docs.renovatebot.com/renovate-schema.json",
+  "extends": [
+    "github>eik-lib/renovate-presets:top-level-module"
+  ]
+}
+```
+
+### Sub modules
+
+Sub modules is __not__ publicly facing modules. They are internal modules which contain code only of interest of top modules and hey should never be used by externals as a dependency.
```
Comment (digitalsadhu) [inline README.md]:
```suggestion
Sub modules are __not__ publicly facing. These are internal modules which contain code only of interest to top level modules and should never be used by external projects and modules as dependencies.
```

File: README.md [hunk 14]
```diff
@@ -1,2 +1,72 @@
...
+
+Sub modules has its dependencies updated as they are published.
```
Comment (digitalsadhu) [inline README.md]:
```suggestion
Sub modules have their dependencies updated as they are published.
```

File: README.md [hunk 15]
```diff
@@ -1,2 +1,72 @@
...
+
+To use the preset for sub modules in a module add the following config to `renovate.json`:
```
Comment (digitalsadhu) [inline README.md]:
```suggestion
To use the preset for sub modules within another module, add the following config to `renovate.json`:
```

File: README.md [hunk 16]
```diff
@@ -1,2 +1,72 @@
...
+  "$schema": "https://docs.renovatebot.com/renovate-schema.json",
+  "extends": [
+    "github>eik-lib/renovate-presets:sub-level-module"
+  ]
+}
+```
+
+## Dev dependencies
+
+Dev dependencies is updated once a month. Both top and sub modules have the same dev dependency preset.
```
Comment (digitalsadhu) [inline README.md]:
```suggestion
Dev dependencies are updated once a month. Both top level and sub modules have the same dev dependency preset.
```

**Review by digitalsadhu — APPROVED** [review]
Nice! I've changed "Top modules" to "Top level modules" in my suggestions. Felt more natural to me but understand if its more a term your coining and would prefer to use "top modules"

**Comment by trygve-lie** [comment]
Yeah. I agree that's better wording.

**Pull request merged**